### PR TITLE
Introduce LearnDocker

### PR DESCRIPTION
**Pull request by josharsh**

### CODE REVIEW

**Comment by ritwik12** [comment]
@josharsh Your branch is not clean, please clean it

**Comment by josharsh** [comment]
Not getting what's wrong. Someone please help!

**Review by ritwik12 — CHANGES_REQUESTED** [review]
Need to clean your branch

**Comment by josharsh** [comment]
Thanks @ritwik12 . Working On it

**Comment by ritwik12** [comment]
This happens when your fork is behind the upstream (main repo where you send PR)

**Comment by ritwik12** [comment]
You need to [sync your fork](https://help.github.com/en/articles/syncing-a-fork)

**Comment by ritwik12** [comment]
This happens when your PR is there and some other changes are pushed to upstream. So your head changes and it becomes merge conflicts.

File: _data/projects/Learning_Object_Oriented_Python.yml [hunk 1]
```diff
@@ -9,6 +12,9 @@ tags:
 upforgrabs:
   name: good first issue
   link: https://github.com/josharsh/Learning-Object-Oriented-Python/labels/good%20first%20issue
+<<<<<<< HEAD
 stats:
   issue-count: 2
   last-updated: '2019-10-01T23:45:51Z'
+=======
+>>>>>>> 3f2aefc8a8560a190bbcabbfbbae51d67ff024a4
```
Comment (ritwik12) [inline _data/projects/Learning_Object_Oriented_Python.yml]: Here remove those >>>>>> and ===== from line 1, 3, 4, 15, 19 and 20

**Review by ritwik12 — CHANGES_REQUESTED** [review]

File: _data/projects/Learning_Object_Oriented_Python.yml (hunk 1 above)
Comment (ritwik12) [reply _data/projects/Learning_Object_Oriented_Python.yml]: @josharsh

File: _data/projects/Learning_Object_Oriented_Python.yml [hunk 2]
```diff
@@ -1,4 +1,3 @@
----
```
Comment (ritwik12) [inline _data/projects/Learning_Object_Oriented_Python.yml]: @josharsh Add it back, you made a change to some other file

**Review by ritwik12 — CHANGES_REQUESTED** [review]

File: _data/projects/LearnDocker.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+site: http://github.com/josharsh/LearnDocker
+tags:
+- oss
+- docker
+- containers
+
+upforgrabs:
+  name: good first issue
+  link: https://github.com/josharsh/LearnDocker/labels/good%20first%20issue
+
```
Comment (ritwik12) [inline _data/projects/LearnDocker.yml:12]: You can remove extra line also @josharsh

File: _data/projects/LearnDocker.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+---
+name: Learn Docker
+desc: >-
+  This repository walks you through the basics of Docker and Containers. The repository contains all the commands for getting started with using Docker!
+site: http://github.com/josharsh/LearnDocker
+tags:
+- oss
+- docker
+- containers
+
```
Comment (ritwik12) [inline _data/projects/LearnDocker.yml:8]: You can remove extra line also @josharsh

**Review by ritwik12 — CHANGES_REQUESTED** [review]

File: _data/projects/Learning_Object_Oriented_Python.yml (hunk 2 above)
Comment (josharsh) [reply _data/projects/Learning_Object_Oriented_Python.yml]: Didn't quite get what happened. I tried resolving the conflicts. I mistakenly used a single commit initially which affected both files. I am stuck now!

Comment (josharsh) [reply _data/projects/Learning_Object_Oriented_Python.yml]: I can't see extra lines added onto the files into my repository, and it looks fine on syntax.

Comment (shiftkey) [reply _data/projects/Learning_Object_Oriented_Python.yml]: > I am stuck now!

I'd just use the web editor to [create a new file](https://help.github.com/en/articles/creating-new-files) against the default branch.

**Comment by ritwik12** [comment]
@josharsh Don't be afraid of Git, you can always go back to make things correct.
Few guidelines for you:

- Always create a branch of the master (here gh-pages) before making commits so that you can always go back by creating another branch of master.
- Never directly use master branch for PRs.

Well as you have already made few mistakes lets make them correct, it will be a good learning for you also.
Follow these steps:

1. As I can see it all happend when you did `Changes Pulled` commit in your branch. 
2. You can easily delete all those commits by doing `git rebase -i HEAD~10`, then you can just remove those lines in which you can see your new commits.
3. Deleting a line in rebase will delete that commit also.
4. After this, you can again add the changes to the file you want to make.


Another thing you can do is to hard reset your branch.

1. Type `git reset --hard origin/master`
2. After this, you can again add the changes to the file you want to make.

Doing any of this will most probably close this PR as when you will reset your branch it will have no changes and without any changes, there is no meaning of PR, so GitHub will close this. But you can click on reopen after making changes.

The easiest thing is to just close this and create new PR, but that way you will not learn and this will trouble you again in the future.

If you have any doubts feel free to ask :)

**Comment by ritwik12** [comment]
@josharsh It looks fine now

**Review by ritwik12 — APPROVED** [review]

**Comment by josharsh** [comment]
@ritwik12  Thanks for the help but I am still in the process to figure out where it went wrong. Please bear with me for the spammed PRs.

**Comment by ritwik12** [comment]
@josharsh It went wrong when you pulled changes, I guess you forgot to pull upstream and pulled your fork from upstream without updating it.

**Comment by josharsh** [comment]
@ritwik12  I got what happened there. Thank you so much for your help! :)

**Review by shiftkey — APPROVED** [review]

**Comment by ritwik12** [comment]
@josharsh Thanks a lot for all your efforts!!

**Pull request merged**